src/main/java/org/springframework/data/elasticsearch/core/DefaultIndexOperations.java

```diff
@@ -256,5 +259,29 @@ public boolean deleteTemplate(DeleteTemplateRequest deleteTemplateRequest) {
 				client -> client.indices().deleteTemplate(deleteIndexTemplateRequest, RequestOptions.DEFAULT).isAcknowledged());
 	}
 
+	@Override
+	public List<IndexInformation> getInformation() {
+		IndexCoordinates indexCoordinates = getIndexCoordinates();
+		GetIndexRequest request = requestFactory.getIndexRequest(indexCoordinates);
+
+		Map<String, Set<AliasData>> aliases = getAliasesForIndex(indexCoordinates.getIndexNames());
+
+		return restTemplate.execute(
+				client -> {
+					GetIndexResponse getIndexResponse = client.indices().get(request, RequestOptions.DEFAULT);
+					List<IndexInformation> indexInformationList = new ArrayList<>();
+
+					for (String indexName : getIndexResponse.getIndices()) {
+						Document settings = requestFactory.settingsFromGetIndexResponse(getIndexResponse, indexName);
+						Document mappings = requestFactory.mappingsFromGetIndexResponse(getIndexResponse, indexName);
+						Set<AliasData> indexAliases = aliases.get(indexName);
+
+						indexInformationList.add(IndexInformation.create(indexName, settings, mappings, indexAliases));
+					}
```

Comment:
Couldn't this whole loop be moved as well into the `RequestFactory`? Something like

```java
List<IndexInformation> indexInformationList = requestFactory.fromGetIndexResponse(getIndexResponse);
```

Reply:
I also didn't really like writing the same thing 3 times.  But shouldn't request factory create only Request objects?
Is that a good place to add a function which returns a list of ```IndexInformation```?

Reply:
fair point. The problem is that the reactive and non-reactive templates have no common base. Now that there are more than just one method for response data it might make sense to add a `ResponseConverter` class next to the `RequestFactory` that contains the methods for processing response data and that is used like the `Requestfactory`is.